scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,136 @@
+require("scripts/globals/settings")
+require("scripts/globals/survival_guide_map")
+
+dsp = dsp or {}
+dsp.survival_guide = dsp.survival_guide or {}
+
+-------------------------------------------------
+-- local functions
+-------------------------------------------------
+
+-- 1 = Original zones
+-- 2 = Zilart Zones
+-- 4 = Chains of Promethia Zones
+-- 8 = Treasures of Aht Urgahn
+-- 16 = Wings of the Goddess
+-- 2048 = Seekers of Auldoin
+local function getEnabledExpansions()
+    -- Original areas and zilart are always available.
+    local returnValue = 1 + 2
+
+    if (ENABLE_COP == 1) then returnValue = returnValue + 4 end
+
+    if (ENABLE_TOAU == 1) then returnValue = returnValue + 8 end
+
+    if (ENABLE_WOTG == 1) then returnValue = returnValue + 16 end
+
+    if (ENABLE_SOA == 1) then returnValue = returnValue + 2048 end
+
+    return returnValue
+end
+
+local function checkForRegisteredSurvivalGuide(player, guide, masks)
+    local group = guide.group
+    local currentZoneId = player:getZoneID()
+    local mask = nil
+    local variableSuffix = ''
+    local hasRegisteredGuide = false
+
+    if (guide.groupIndex == 32) then
+        mask = masks[group + 4]
+        variableSuffix = 'a'
+
+        -- This turns negative in the database for int32 max
+        hasRegisteredGuide = mask == (guide.groupMask * -1)
+    else
+        mask = masks[group]
+        hasRegisteredGuide = (bit.band(mask, guide.groupMask) ~= 0)
+    end
+
+    if (not hasRegisteredGuide) then
+        player:PrintToPlayer(
```

Comment:
We don't use PrintToPlayer for this. That message will either be part of a cs event, a messageSpecial, or another type of message in client accessed by its ID.

Reply:
That makes sense, can you point me towards where I can figure out the appropriate ID? The only CSIDs I could fine for Survival Guides are 8500 and 8501 and they both say the same thing in the UI (basically the first line to go into the teleportation sub-menu)

Reply:
all strings zone specific can be located via polutils. This type of things is likely to be the zone specific `messageSpecial()`. In menu, I believe the ones you don't have yet don't even show to be selected (so no message at all).

Reply:
Cool, I got the tool and am going through it. Will probably take a while (day or 2) to get all of the IDs together since they are different per zone.